tests/unit/s2n_kex_with_kem_test.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tests/s2n_test.h"
+
+#include "tls/s2n_cipher_suites.h"
+#include "tls/s2n_client_key_exchange.h"
+#include "tls/s2n_server_key_exchange.h"
+#include "tls/s2n_kex.h"
+#include "tls/s2n_kex_data.h"
+#include "tls/s2n_kem.h"
+#include "tls/s2n_tls.h"
+
+
+#include "utils/s2n_safety.h"
+
+#define KEM_ID 0xab
+#define TEST_PUBLIC_KEY_LENGTH 0x0002
+const uint8_t TEST_PUBLIC_KEY[] = {0x02, 0x02};
+
+#define TEST_PRIVATE_KEY_LENGTH 3
+const uint8_t TEST_PRIVATE_KEY[] = {0x03, 0x03, 0x03};
+
+#define TEST_SHARED_SECRET_LENGTH 4
+const uint8_t TEST_SHARED_SECRET[] = {0x04, 0x04, 0x04, 0x04};
+
+#define TEST_CIPHERTEXT_LENGTH 0x0005
+const uint8_t TEST_CIPHERTEXT[] = {0x05, 0x05, 0x05, 0x05, 0x05};
+
+// The server send key message is defined as:
+// 1 byte for KEM ID
+// 2 bytes for public key length
+// the public key
+const int TEST_SERVER_SEND_KEY_MESSAGE_LENGTH = 1 + 2 + 2;
```

Comment:
Please use TEST_PUBLIC_KEY_LENGTH

Reply:
Changed these all to use sizeof the respective part.